### ISSUE TITLE
Turn `migratePresets` into a non-self-hosted property

### ISSUE THREAD

**Issue by astellingwerf** [renovatebot/renovate#16605]
### What would you like Renovate to be able to do?

The property `migratePresets` can be used to migrate presets (shocker, I know!). Today, the property is a self-hosted property. I'd like to suggest that I'd would be of more use if the property could also be set in a repo or preset context. More specifically: whenever implementable, it should be possible for a "legacy" preset to include a `migratePresets` declaration about itself and its replacement, in order to auto-discover the new prefered situation.

### If you have any ideas on how this should be implemented, please tell us here.

Not really.

### Is this a feature you are interested in implementing yourself?

Maybe

**Comment by viceice** [comment]
doesn't make sense, as you still need to run renovate on it.

so the request would be to allow migrating presets instead. so please update issue

**Comment by astellingwerf** [comment]
That's what `configMigration=true` achieves if `migratePresets` is also set. I don't see a reason to update the issue.

**Comment by rarkins** [comment]
So I think ideally it would have been called something like "remapPresets" because it doesn't migrate anything itself per se. But now that we have it, is it work renaming it?

**Comment by HonkingGoose** [comment]
If the current name is confusing we should pick a better name, even if that means more work. Good names help to understand what the bot is going to do. 😄

**Comment by rarkins** [comment]
Yes, let's have a separate issue. We need to remember to migrate self-hosted config in this case

**Comment by HonkingGoose** [comment]
> Yes, let's have a separate issue. 

Done, see: 

- #19291 

> We need to remember to migrate self-hosted config in this case

Can you tell me where that todo belongs? I don't understand if it's for _this_ issue or the _new separate_ issue. 🙈

**Comment by rarkins** [comment]
Within the same issue

**Comment by Finkregh** [comment]
This would be _really_ helpful, as it would allow rename files:

- enable `configMigration` in the "old" file (yes, might lead to other PRs being created, but that's OK in my case)
- set `migratePresets` in the "old" file to reference the new/renamed file
